### PR TITLE
feat: mentions support cssVar

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | mentions support css variable theme |
| 🇨🇳 Chinese | 提及支持 css 变量主题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at b1ea7ac</samp>

This pull request refactors the Mentions component to use CSS variables for theming and customization. It introduces a new file `cssVar.ts` that exports a hook for registering the variables, and modifies the component and style files to use the hook and the utility functions. It also removes some unused code and adds a new style property.

### <samp>🤖 Generated by Copilot at b1ea7ac</samp>

*  Add hooks for registering and applying CSS variables to Mentions component and its dropdown menu ([link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-70d5a1401a2836d63ffd091fbcd5e9f43de05c2f404265a51a366862e1e157b6R21-R22), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-11c809afdfac767e137e2597f75933821d9335bc0d36ff954da66c743bc0588eR1-R4), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceR219-R226), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceL225-R234))
*  Use `useCSSVarCls` hook to generate and add a CSS variable prefix class to the component's root and dropdown elements ([link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-70d5a1401a2836d63ffd091fbcd5e9f43de05c2f404265a51a366862e1e157b6L157-R161), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-70d5a1401a2836d63ffd091fbcd5e9f43de05c2f404265a51a366862e1e157b6R174), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-70d5a1401a2836d63ffd091fbcd5e9f43de05c2f404265a51a366862e1e157b6L184-R189))
*  Use `useCSSVar` hook to wrap the component with a style tag that defines the CSS variables ([link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-70d5a1401a2836d63ffd091fbcd5e9f43de05c2f404265a51a366862e1e157b6L157-R161), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-70d5a1401a2836d63ffd091fbcd5e9f43de05c2f404265a51a366862e1e157b6L194-R199))
*  Add `itemPaddingVertical` property to `MentionsToken` type and calculate it from other token properties ([link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceL32-R36), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceL52-R58), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceR219-R226))
*  Use `unit` function to convert token values to CSS unit values when generating CSS variables and styles ([link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceL95-R98), [link](https://github.com/ant-design/ant-design/pull/45796/files?diff=unified&w=0#diff-9600e9fcd22b061804bbc1129620d20434a9c89a29b42a202c824aadda4c6eceL179-R180))

### CODE REVIEW

File: components/mentions/style/index.ts [hunk 1]
```diff
@@ -49,12 +53,9 @@ const genMentionsStyle: GenerateStyle<MentionsToken> = (token) => {
     borderRadius,
     borderRadiusLG,
     boxShadowSecondary,
+    itemPaddingVertical,
   } = token;
 
-  const itemPaddingVertical = Math.round(
-    (token.controlHeight - token.fontSize * token.lineHeight) / 2,
```
Comment (MadCcc) [inline components/mentions/style/index.ts:55]: 这里也可以用 fontHeight 换一下了，不需要加 token 了，0.5 的小数问题不大

File: components/mentions/style/index.ts [hunk 2]
```diff
@@ -29,7 +30,10 @@ export interface ComponentToken extends SharedComponentToken {
   controlItemWidth: number;
 }
 
-type MentionsToken = FullToken<'Mentions'> & SharedInputToken;
+type MentionsToken = FullToken<'Mentions'> &
+  SharedInputToken & {
+    itemPaddingVertical: string | number;
```
Comment (MadCcc) [inline components/mentions/style/index.ts:35]: 仔细一想声明在这里好像也行。。

File: components/mentions/style/index.ts (hunk 1 above)
Comment (MadCcc) [inline components/mentions/style/index.ts:55]: 这里应该替换一下 fontHeight 就好了，`/2` 应该不需要取整，取整后反而会让 item 的高度和 controlHeight 不一样

Comment (kiner-tang) [reply components/mentions/style/index.ts:55]: 好的，那我把 取整逻辑也去掉

Comment (kiner-tang) [reply components/mentions/style/index.ts:55]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**